Configurable and interactive control whether to act on 'file at point'…

Gallipo opened the pull request:
… or 'marked files'

Gallipo committed:
Configurable and interactive control wheter to act on 'file at point' or 'marked files'

Gallipo committed:
deleted unnessesary message

Gallipo committed:
improved handling of mixed MIME-typ selections

vifon commented on dired-rifle.el:
```diff
@@ -94,25 +104,32 @@ With `\\[universal-argument] \\[universal-argument]' the output
-                    program-number
-                    output-buffer)))))
+          (path (if (or (and (equal arg 0) (not dired-rifle-use-marked-files))
+			(and (not (equal arg 0)) dired-rifle-use-marked-files))
```
This conditional could use a refactor. Maybe compare `(equal arg 0)` to `dired-rifle-use-marked-files`, essentially making it invert the condition?

Gallipo replied:
Good point. I never saw that possibility

vifon commented on dired-rifle.el:
```diff
@@ -94,25 +104,32 @@ With `\\[universal-argument] \\[universal-argument]' the output
+			(and (not (equal arg 0)) dired-rifle-use-marked-files))
+		    (dired-get-marked-files)
+		  (dired-get-filename)
+		  )))
```
Let's move these hanging parens to the previous line.

Gallipo replied:
sure

vifon commented on dired-rifle.el:
```diff
@@ -94,25 +104,32 @@ With `\\[universal-argument] \\[universal-argument]' the output
+		  (dired-get-filename)
+		  )))
+      (dolist (p (flatten-tree path))
+	(let ((program-number (if (consp arg)
```
I see you've been using tabs for indentation. Let's stick to spaces in this project. You can use `M-x untabify`.

Gallipo replied:
That wasn't intentional, sorry. To be honest, I'm new to this. I will try my best to improve. Thanks for your input.

Gallipo committed:
Refactoring condition

vifon commented:
Apologies for the delay. I'll try to take a look this weekend.

Gallipo commented:
I didn't mean to stress you. I'm a little unsure how to communicate on this platform. Please take the time you need.

vifon commented on dired-rifle.el:
```diff
@@ -42,11 +42,20 @@
+  :type '(choice
+          (const :tag "single-file" nil)
+          (const :tag "marked-files" t))
+  )
```
I must have missed it before, let's put this parenthesis on the previous line. This is the style most commonly used in Lisps.

vifon commented on dired-rifle.el:
```diff
@@ -42,11 +42,20 @@
   (append (when rifle-config
             (list "-c" (expand-file-name rifle-config)))
-          args))
+          (flatten-tree args)))
```
What's the purpose of `flatten-tree` here? In which scenario is `args` not a flat list?

Gallipo replied:
I do not remember. Seems like I was thinking it is necessary, but obviously it is not.

vifon commented on dired-rifle.el:
```diff
@@ -94,25 +104,30 @@ With `\\[universal-argument] \\[universal-argument]' the output
+          (path (if (equal (equal arg 0) dired-rifle-use-marked-files)
+                    (dired-get-filename)
+                  (dired-get-marked-files))))
+      (dolist (p (flatten-tree path))
```
I've been thinking… Shouldn't we handle the whole argument list in a single `rifle` call instead of calling `rifle` once for each argument? Is this a conscious decision on your part for some reason?

Gallipo replied:
Yes, if you mix MIME-Type in your marking (e.g. .pdf and .txt), the single `rifle` call can open the correct programs. 
I'm working on grouping MIME-Types so that all files of the same type get opened by one instance of a program, but I'm not done yet.

vifon replied:
That's true, though it's also consistent with the behavior of the original rifle. If we were to fix it, I'd rather fix it there.

vifon commented on dired-rifle.el:
```diff
@@ -94,25 +104,30 @@ With `\\[universal-argument] \\[universal-argument]' the output
+          (path (if (equal (equal arg 0) dired-rifle-use-marked-files)
+                    (dired-get-filename)
+                  (dired-get-marked-files))))
+      (dolist (p (flatten-tree path))
```
Again, what's the purpose of `flatten-tree` here? Is this to make the output of `(dired-get-filename)` into a one-element list? We can handle it above by doing that explicitly.

vifon reviewed (commented):
Hello again. Sorry for the delay on my part, it was a busy period for me. Don't worry about your communication and contributions on Github, you're doing great so far, that's a quality code, even if it needs some polishing before merging (that's perfectly normal).

In a few minutes I'll push my proposed changes addressing my remarks, tell me what you think then.

vifon committed:
Coding style cleanup

vifon committed:
Cleanup the marked files handling

All the files are now handled by a single rifle call, the format of
"file-paths" is now also more consistent, it's always a list, possibly
with only one element.

vifon commented:
Done. Please tell me what you think. If you're fine with my changes, I believe it's ready for merging.

Gallipo commented:
Looks good to me.
I will keep on working on the matter of mixed-mime-type-selections, but that will (hopefully) result in another pull request, when/if ready.

vifon commented:
Great, I'm merging then! Thank you!
